plugin/cr/src/main/java/org/opencds/cqf/ruler/cr/KnowledgeArtifactProcessor.java

```diff
@@ -1712,6 +1725,19 @@ public MetadataResource revise(HapiFhirRepository hapiFhirRepository, MetadataRe
 
 		return resource;
 	}
+
+	private static Library getRootSpecificationLibrary(List<BundleEntryComponent> bundleEntries) {
+		Optional<BundleEntryComponent> rootSpecLibraryEntry = bundleEntries.stream().filter(entry ->
```

Comment:
We should probably take this even a step further and make more specific. Specifically, per the [us-ph-specification-library profile in the electronic Case Reporting IG](http://hl7.org/fhir/us/ecr/STU2.1.1/StructureDefinition-us-ph-specification-library.html), the root specification library will also have the following useContexts:
`{
        "code": {
          "system": "http://hl7.org/fhir/us/ecr/CodeSystem/us-ph-usage-context-type",
          "code": "reporting"
        },
        "valueCodeableConcept": {
          "coding": [ {
            "system": "http://hl7.org/fhir/us/ecr/CodeSystem/us-ph-usage-context",
            "code": "triggering"
          } ]
        }
      }, {
        "code": {
          "system": "http://hl7.org/fhir/us/ecr/CodeSystem/us-ph-usage-context-type",
          "code": "specification-type"
        },
        "valueCodeableConcept": {
          "coding": [ {
            "system": "http://hl7.org/fhir/us/ecr/CodeSystem/us-ph-usage-context",
            "code": "program"
          } ]
        }
      }`
      
So, we can add checks to confirm that it has those two use contexts as more specific way of identifying the library.

Reply:
Done!